checker/checker.go

```diff
@@ -75,14 +75,18 @@ type Checker struct {
 		sync.RWMutex
 		detail *check.Results
 	}
+	errCnt  int64
```

Comment:
Is it make sense to restrict the total length of the errors or the total numbers of error + warn instead of set two different count separately. e.g. if we think be default we only return 20 messages, then if errors + warns <= 20, we needn't truncate any messages.

Reply:
I think error has a higher priority than warning, so if user gets 20 warnings and 1 errors we should make sure we will display that errors.

Reply:
we could loop the results twice to make sure errors have higher priority 🤔 I'm OK with current two settings, let's wait glorv

Reply:
I'm ok, too. I just wonder if this implementation is user-friendly enough to use since the goal is to solve the message too long issue.

BTW, I think maybe there are some cases the user just wants to emit the warning message, thus the `--warn` parameter can be used.